Improve guesses/stopwatch display and add button to reveal new hint

JEANPRBT opened the pull request:
This PR improves the display of the number of guesses and of the stopwatch, which allows to add a button to reveal a new hint, avoiding the hassle of guessing a random celebrity. See picture below for new display, both on desktop and mobile devices. It should be merged after PR #164 since it already merged the major changes of this branch. 
Closes #176 
Closes #169 

<img width="1470" alt="new_display_desktop" src="https://github.com/roxannecvl/whokipedia/assets/66010389/d897f05c-0be6-4125-8ef7-8fe4532d863c">
<img width="1470" alt="new_display_mobile" src="https://github.com/roxannecvl/whokipedia/assets/66010389/e36078bb-1e0b-4325-83c7-3ebf5f8a901d">

See background color slightly changed upon game finish. 

<img width="1470" alt="new_display_game_over" src="https://github.com/roxannecvl/whokipedia/assets/66010389/466a87f2-c9d1-4244-be6f-8c73cb5c312f">

roxannecvl reviewed (commented):
I haven't looked at the code yet but I have a few comments already. 
It looks prettier in general but I don't like that you don't know how many guesses you have in total and that you should approximate according to the loading bar.
The "+" button is a great idea but I don't think that it is clear that it gives you a new hint and counts as a new guess. Also it shouldn't be there when the game is won I think.

JEANPRBT commented:
@roxannecvl I added the total number of guesses on the right so that it is always visible. Also, I added the `+` button feature in the rules so that the player can understand what is it doing. Finally, this same button is now hidden when game is over.
